### PR TITLE
UserProfileOverlay Mode Switcher Implementation

### PR DESCRIPTION
**Pull request by EVAST9919**
screenshot-460

Also fixes a lot of possible nulls in rank graph and user request

### CODE REVIEW

File: osu.Game/Overlays/UserProfileOverlay.cs [hunk 1]
```diff
@@ -179,16 +216,73 @@ private void userLoadComplete(User user)
...
+                    sectionTabs.AddItem(sec);
                 }
             }
+
+            string playMode;
+            switch (user.PlayMode)
+            {
+                default:
+                case "osu":
+                    playMode = "osu!";
```
Comment (peppy) [inline osu.Game/Overlays/UserProfileOverlay.cs]: these should be retrieved from RulesetStore. we don't want to have string constants across the code base

File: osu.Game/Overlays/UserProfileOverlay.cs [hunk 2]
```diff
@@ -179,16 +216,73 @@ private void userLoadComplete(User user)
...
+                    break;
+            }
+            modeTabs.Current.Value = playMode;
+            modeTabs.FadeIn(200, Easing.Out);
+
+            modeTabs.Current.ValueChanged += updateMode;
         }
 
-        private class ProfileTabControl : PageTabControl<ProfileSection>
+        private void updateMode(string newMode)
```
Comment (peppy) [inline osu.Game/Overlays/UserProfileOverlay.cs]: string? please no

**Comment by peppy** [comment]
play mode should be tracked as a RulesetInfo, not an enum or string.

Also you may want to consider making it a bindable, rather than passing it down using a weird combination of properties and methods.

**Comment by EVAST9919** [comment]
A lot stuff should be done to implement this feature properly. Also i'll rather split it in a smaller pieces for consistency.

**Pull request closed without merging**